frontend/packages/console-dynamic-plugin-sdk/src/extensions/user-preferences.ts

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { Extension, ExtensionDeclaration, CodeRef } from '../types';
+
+export enum UserPreferencesInputType {
+  dropdown = 'dropdown',
+  checkbox = 'checkbox',
+}
+
+export type DropdownOption = {
+  type: UserPreferencesInputType.dropdown;
+  options: {
+    value: string;
+    label: string;
+  }[];
+};
+export type CheckboxOption = {
+  type: UserPreferencesInputType.checkbox;
+  trueValue: string;
+  falseValue: string;
+};
+export type UserSettingInputOptions = DropdownOption | CheckboxOption;
```

Comment:
We should make a decision to call all these classes (and files, fields, ..) `UserPreferences...` or `UserSettings...`, and try to avoid mixing both. I'm fine with UserPreferences which is imho a more common term and navigation entry and page title calls this "User preferences".

Imho we should keep just `useUserSettings` and create a new issue to rename all useUserSettings hooks in another PR (to keep the focus here on the new components and still cleanup these later).

@christianvogt @invincibleJai fine with that or do you have any preference (hihi) ?

Reply:
Ah my idea was to call the individual preference field, because Aritra call the field options "inputOptions".

We have three levels with really similar name where it would be good to have as clean as possible names:

1. Groups
2. Items (or fields)
3. Item input definitions (or option) for each item (or field or option)